### PR TITLE
Add `fundingFee` to `IncomingPayment.ReceivedWith.NewChannel`

### PR DESCRIPTION
**Pull request by robbiehanson**
With the addition of DualFunding, there are now 2 separate fees as shown in `SwapInEvents.Accepted`:

```kotlin
data class Accepted(
  val requestId: ByteVector32,
  val fundingFee: Satoshi, // L1 fees paid to miner
  val serviceFee: MilliSatoshi // L2 fees paid to lightning service provider
) : SwapInEvents
```

However, the fundingFee wasn't exported to `IncomingPayment.ReceivedWith.NewChannel`. Only the serviceFee was included. So, for example, a user sends 15,000 sat to their wallet, and we show:

- received: 11,931 sat
- fees: 3,000 sat

This PR adds both fees to the class:

```kotlin
data class NewChannel(
  val id: UUID,
  override val amount: MilliSatoshi,
  val serviceFee: MilliSatoshi,
  val fundingFee: Satoshi = 0.sat,
  val channelId: ByteVector32?
) : ReceivedWith() {
  override val fees: MilliSatoshi = serviceFee + fundingFee.toMilliSatoshi()
}
```

This allows the UI to properly display each fee to the user. And we extract the fundingFee here:

```kotlin
// oldState is WaitForFundingSigned
when {
  !oldState.localParams.isInitiator -> {
    val amount = oldState.fundingParams.localAmount.toMilliSatoshi() + oldState.remotePushAmount - oldState.localPushAmount
    val fundingFee = oldState.fundingTx.localFees(oldState.fundingParams)
    val localInputs = oldState.fundingTx.localInputs.map { OutPoint(it.previousTx, it.previousTxOutput) }.toSet()
    actions + ChannelAction.Storage.StoreIncomingAmount(amount, fundingFee, localInputs, oldState.channelOrigin) + channelCreated
  }
  else -> actions + channelCreated
}
```

### CODE REVIEW

**Review by pm47 — APPROVED** [review]
In theory the funding fee should probably be in `PleaseOpenChannelOrigin`, so you don't have to change `StoreIncomingAmount`, but it would require updating eclair TLV too. In the meantime this is fine, and we will revisit when we adapt the pay-to-open mechanism to work with splices.

Please don't merge right away, I'd like to avoid conflicts with the monster PR #391.

**Comment by pm47** [comment]
I took advantage on the rebase on eclair to modify the channel origin tlv, and implemented on top of this PR in 212bec3. Feel free to cherry-pick that change.

Btw I have also fixed the issue on eclair which caused channel opening to be aborted due to a fee too high. I suppose you were using a workaround for that?

**Comment by robbiehanson** [comment]
> I took advantage on the rebase on eclair to modify the channel origin tlv, and implemented on top of this PR in https://github.com/ACINQ/lightning-kmp/commit/212bec3abb121a6a2b43a7601bda1b6bc73787d5. Feel free to cherry-pick that change.

Awesome! Done. And merged master to fix the conflict.

> Btw I have also fixed the issue on eclair which caused channel opening to be aborted due to a fee too high. I suppose you were using a workaround for that?

Yes, dpad told me to increase the hard-coded values as a temporary workaround:

```kotlin
- input.send(RequestChannelOpen(Lightning.randomBytes32(), wallet, maxFeeBasisPoints = 100, maxFeeFloor = 3_000.sat)) // 100 bips = 1 %
+ input.send(RequestChannelOpen(Lightning.randomBytes32(), wallet, maxFeeBasisPoints = 1000000, maxFeeFloor = 1_000_000_000.sat)) // 100 bips = 1 %
```

File: src/commonTest/kotlin/fr/acinq/lightning/wire/OpenTlvTestsCommon.kt
Comment (pm47) [inline src/commonTest/kotlin/fr/acinq/lightning/wire/OpenTlvTestsCommon.kt:69]: Oops, this needs to be uncommented, my bad!

Comment (robbiehanson) [reply src/commonTest/kotlin/fr/acinq/lightning/wire/OpenTlvTestsCommon.kt]: Done in [d9eb1ad](https://github.com/ACINQ/lightning-kmp/pull/395/commits/d9eb1adf48c6daa6708da13273b8bac64eabbc86)

**Review by pm47 — APPROVED** [review]

**Pull request merged**
